javascript/hangman.js

```diff
@@ -1,55 +1,88 @@
 class Hangman {
   constructor(words) {
     this.words = words;
-    // ... your code goes here
+    //Iteration 1: The game logic
+    //Hangman Class
+    this.secretWord = this.pickWord();
+    this.letters = [];
+    this.guessedLetters = "";
+    this.errorsLeft = 10;
   }
-
+  //The Hangman methods
   pickWord() {
-    // ... your code goes here
+    return this.words[Math.floor(Math.random() * this.words.length)];
   }
 
   checkIfLetter(keyCode) {
-    // ... your code goes here
+    return keyCode < 65 && keyCode > 90 ? false : true;
   }
 
   checkClickedLetters(letter) {
-    // ... your code goes here
+    return this.letters.indexOf(letter) == -1;
   }
 
   addCorrectLetter(letter) {
-    // ... your code goes here
+    if (!this.guessedLetters.includes(letter)) this.guessedLetters += letter;
   }
 
   addWrongLetter(letter) {
-    // ... your code goes here
+    if (!this.secretWord.includes(letter)) this.errorsLeft--;
   }
 
   checkGameOver() {
-    // ... your code goes here
+    return this.errorsLeft === 0;
   }
```

Comment:
Muy buena sintesis, aqui mismo podriamos retornar `!this.errorsLeft`

Reply:
Gracias!!